keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -0,0 +1,962 @@
+-->
+
+### Troubleshooting
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+- 2023-01-19: Initial KEP
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+1. Add `PreemptLowerPriorityWithoutViolatePDB` as an option in the `PreemptionPolicy`. 
+   When the preemptor is set to `PreemptLowerPriorityWithoutViolatePDB`, the pods that would violate 
+   PDBs will be excluded when selecting victims. However, if we want to guarantee some pods' PDBs, 
+   we need to modify the `PreemptionPolicy` for all other pods to `PreemptLowerPriorityWithoutViolatePDB`. 
+   The cost of this operation may be relatively large, and it's a little strange. Also, the more natural 
+   place to describe PDB behavior is in the PDB spec.
+
+2. Add `NotAllowViolatePDB` as an option in the `PreemptionPolicy`. When the pods are set to `NotAllowViolatePDB`, 
```

Comment:
Same question as the one above.

They actually sound like the same alternative, but with slightly different names.

Reply:
NO. The option1 is the policy for the preemptor, option2 is the policy for the victim. Additional explanations have been added to the design document.